src/synthesizer/mod.rs

```diff
@@ -0,0 +1,335 @@
+use crate::ir::conditions::ConditionIr;
+use crate::ir::mappings::MappingInstruction;
+use crate::ir::reference::Origin;
+use crate::ir::reference::PseudoParameter;
+use crate::ir::CloudformationProgramIr;
+use crate::parser::lookup_table::MappingInnerValue;
+use crate::parser::resource::ResourceValue;
+use crate::parser::sub::{sub_parse_tree, SubValue};
+use crate::CloudformationParseTree;
+use std::collections::HashMap;
+use voca_rs::case::camel_case;
+
+// TODO - remove parse_tree
+pub fn synthesize(parse_tree: CloudformationParseTree, ir: CloudformationProgramIr) {
+    for import in ir.imports {
+        println!(
+            "import * as {} from '{}';",
+            import.name,
+            import.path.join("/")
+        )
+    }
+
+    println!("export interface NoctStackProps extends cdk.StackProps {{");
+    for param in ir.constructor.inputs {
+        println!(
```

Comment:
For later, but wondering if we should have some type of indent helper that tracks how many indents to put at the start of a line.

Reply:
that would be nice. As of right now i just run programs like:

`cargo run example.json | tsfmt --stdin > example.ts`

Which works just as well. Will think about this in future iterations though.